### ISSUE TITLE
vault_consul_secret_backend_role does not support consul acl roles

### ISSUE THREAD

**Issue by BrandonIngalls** [hashicorp/terraform-provider-vault#951]
### Terraform Version

```
[~]$ terraform -v  
Terraform v0.14.5
```

### Affected Resource(s)

* [vault_consul_secret_backend_role](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/consul_secret_backend_role)

### Terraform Configuration Files

```hcl
terraform {
  backend "consul" {
    scheme       = "http"
    address      = "127.0.0.1:8500"
    access_token = "consul-token"
    path         = "terraform/example/state"
  }

  required_providers {
    vault = {
      source  = "hashicorp/vault"
      version = "~> 2.18.0"
    }
    consul = {
      source  = "hashicorp/consul"
      version = "~> 2.11.0"
    }
  }
}

# =============================================================================

locals {
  time = {
    seconds = 1
    minutes = 60 * 1
    hours   = 60 * 60 * 1
    days    = 24 * 60 * 60 * 1
  }
}

# =============================================================================

provider "vault" {
  address = "http://127.0.0.1:8200"
  token   = "root-token"
}

provider "consul" {
  scheme = "http"
  address = "127.0.0.1:8500"
  token = "consul-token"
}

# =============================================================================

resource "consul_acl_policy" "acl_ro" {
  name  = "acl-ro"
  rules = "acl = \"read\""
}

resource "consul_acl_policy" "node_ro" {
  name  = "node-ro"
  rules = "node_prefix \"\" { policy = \"read\" }"
}

resource "consul_acl_policy" "service_ro" {
  name  = "service-ro"
  rules = "service_prefix \"\" { policy = \"read\" }"
}

# =============================================================================

resource "consul_acl_role" "operator_ro" {
  name        = "operator-ro"
  description = "human read-only role for IT members"

  policies = [
    consul_acl_policy.acl_ro.id,
    consul_acl_policy.node_ro.id,
    consul_acl_policy.service_ro.id,
  ]
}

resource "consul_acl_role" "user_ro" {
  name        = "user-ro"
  description = "human read-only role for non IT"

  policies = [
    consul_acl_policy.node_ro.id,
    consul_acl_policy.service_ro.id,
  ]
}

# =============================================================================

resource "vault_consul_secret_backend" "main" {
  path        = "consul"
  description = "Manages the Consul backend"

  address = "127.0.0.1:8500"
  token   = "consul-token"

  max_lease_ttl_seconds = 1 * local.time.days
}

resource "vault_consul_secret_backend_role" "operator_ro" {
  name    = "operator-ro"
  backend = vault_consul_secret_backend.main.path

  # NOTE: I'd like to use `roles` instead of individual policies
  # roles = ["operator-ro"]
  policies = [
    "acl-ro",
    "node-ro",
    "service-ro",
  ]

  max_ttl = 4 * local.time.hours
  ttl     = 30 * local.time.minutes
}

resource "vault_consul_secret_backend_role" "user_ro" {
  name    = "user-ro"
  backend = vault_consul_secret_backend.main.path

  # NOTE: I'd like to use `roles` instead of individual policies
  # roles = ["user-ro"]
  policies = [
    "node-ro",
    "service-ro",
  ]

  max_ttl = 4 * local.time.hours
  ttl     = 30 * local.time.minutes
}
```

### Expected Behavior

`vault_consul_secret_backend_role` should have a `roles = list(string)` option, this would support [Consul's built-in](https://www.consul.io/api-docs/acl/roles) **acl role** feature.

### Actual Behavior

```bash
[~]$ terraform apply

Error: Missing required argument

  on main.tf line 97, in resource "vault_consul_secret_backend_role" "operator_ro":
  97: resource "vault_consul_secret_backend_role" "operator_ro" {

The argument "policies" is required, but no definition was found.


Error: Unsupported argument

  on main.tf line 102, in resource "vault_consul_secret_backend_role" "operator_ro":
 102:   roles = ["operator-ro"]

An argument named "roles" is not expected here.
```

### Steps to Reproduce

1. start dev consul & vault servers
    ```bash
    [~]$ vault server -dev -dev-root-token-id=root-token
    [~]$ consul agent -dev -hcl='acl = {enabled = true default_policy = "deny" enable_token_persistence = true tokens { master = "consul-token" agent = "consul-token"}}'
    ```
2. `terraform apply`

**Comment by BrandonIngalls** [comment]
It looks like Vault does not support Consul roles, so it's not an issue with the terraform provider.

This feature request is currently blocked by hashicorp/vault#10752

**Comment by benashz** [comment]
Fixed in #1366
